integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala

```diff
@@ -402,7 +402,7 @@ class CarbonScanRDD(
       // one query id per table
       model.setQueryId(queryId)
       // get RecordReader by FileFormat
-      val reader: RecordReader[Void, Object] = inputSplit.getFileFormat match {
+      var reader: RecordReader[Void, Object] = inputSplit.getFileFormat match {
```

Comment:
reader can be val...is there any specific purpose for making it var?

Reply:
reader will be set null in closeReader() method to reduce memory occupation when using coalesce, otherwise there will be lots of reader instances in memory.